cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1023,17 +1023,21 @@ CacheAllocator<CacheTrait>::acquire(Item* it) {
 
   SCOPE_FAIL { stats_.numRefcountOverflow.inc(); };
 
-  // TODO: do not block incRef for child items to avoid deadlock
-  auto failIfMoving = getNumTiers() > 1 && !it->isChainedItem();
-  auto incRes = incRef(*it, failIfMoving);
-  if (LIKELY(incRes == RefcountWithFlags::incResult::incOk)) {
-    return WriteHandle{it, *this};
-  } else if (incRes == RefcountWithFlags::incResult::incFailedEviction){
-    // item is being evicted
-    return WriteHandle{};
-  } else {
-    // item is being moved - wait for completion
-    return handleWithWaitContextForMovingItem(*it);
+  while (true) {
```

Comment:
Inside this spin loop, it is a good practice to use pause instructions. Actually, folly already has platform-agnostic function ` asm_volatile_pause()`. Furthermore, there is a `folly::detail::Sleeper` abstraction, but the problem is that it is not a public interface.

Reply:
I don't think we need to pause here - we take a lock on each iteration (and possibly drop it) so we already sleep on that. Moreover, it's  a very rare case for this loop to be executed more than once - in our benchmarks this was happening on average once per 100M ops